src/api/services/routing-auth-service.ts

```diff
@@ -0,0 +1,28 @@
+import {Injectable} from '@angular/core';
+import {CanActivate, Router, ActivatedRouteSnapshot, RouterStateSnapshot} from '@angular/router';
+import {RoutingService} from './routing-service';
+import _ from 'lodash';
+
+@Injectable()
+export class RoutingAuthService implements CanActivate {
+    constructor(private router: Router, private rutingService: RoutingService) {}
+
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
+        return this.isRouteAllow(state.url);
+    }
+
+    public isRouteAllow(url: string): boolean {
+        let isRouteLoaded = _.chain(this.rutingService.currentMenu)
+            .flatMap(item => item.menuItems)
+            .some(['url', url])
+            .value();
```

Comment:
I thought about it, but right now I rather to have their own service because in the future we could have check for other stuff like User Roles or something else and also, that's how the angular tutorial does it :)